File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        queue = deque()
        if root:
            queue.append(root)
        result = list()
        while len(queue) > 0:
            temp = deque()
            next_level = list()

            for node in queue:
           
                next_level.append(node.val)
                if node.right:
                    temp.append(node.right)
                if node.left:
                    temp.append(node.left)
            
            result.append(next_level[::-1])
            queue = temp

        return result

File: Data Structures & Algorithms/insert-into-a-binary-search-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* insertIntoBST(TreeNode* root, int val) {
        TreeNode* curr = root;
        if (root == nullptr){
            return new TreeNode(val);
        }
        while (true){

            if ( curr->left == nullptr and val < curr->val ){
                curr->left = new TreeNode(val);
                break;
            }
            else if (curr->right == nullptr and val > curr->val){
                curr->right = new TreeNode(val);
                break;
            }
            else if (curr->left != nullptr and val < curr->val){
                curr = curr->left;
            }
            else if (curr->right != nullptr){
                curr = curr->right;
            }
            else{
                break;
            }
        }

        return root;
        
    }
};